src/common/WorkQueue.cc

```diff
@@ -39,7 +39,7 @@ ThreadPool::ThreadPool(CephContext *cct_, string nm, string tn, int n, const cha
     ioprio_class(-1),
     ioprio_priority(-1),
     _num_threads(n),
-    last_work_queue(0),
+    last_work_queue(-1),
```

Comment:
instead of making the first iteration an exception, i'd suggest follow change:
```diff
diff --git a/src/common/WorkQueue.cc b/src/common/WorkQueue.cc
index b74763901b..7c91dda618 100644
--- a/src/common/WorkQueue.cc
+++ b/src/common/WorkQueue.cc
@@ -117,9 +117,8 @@ void ThreadPool::worker(WorkThread *wt)
       int tries = work_queues.size();
       bool did = false;
       while (tries--) {
-       last_work_queue++;
+       wq = work_queues[last_work_queue++];
        last_work_queue %= work_queues.size();
-       wq = work_queues[last_work_queue];

        void *item = wq->_void_dequeue();
        if (item) {
```